_posts/2017-12-08-totw-131.md

```diff
@@ -0,0 +1,135 @@
+---
+title: "Tip of the Week #131: Special Member Functions and `= default`"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/131
+type: markdown
+---
+
+Originally posted as totw/131 on 2017-03-24
+
+By James Dennett [(jdennett@google.com)](mailto:jdennett@google.com)
+
+Since the beginning of time, C++ has supported compiler-declared versions of
+some so-called *special member functions*: the default constructor, destructor,
+copy constructor and copy assignment operators. C++11 added move construction
+and move assignment to the list, and added syntax (`=default` and `=delete`) to
+give control over when those defaults are declared and defined.
+
+## What Does `=default` Do, and Why Would We Use It?
+
+Writing `=default` is our way to tell the compiler "do what you would normally
+have done for this special member function". Why might we want to do this
+rather than writing an implementation by hand or leaving the compiler to
+declare one for us?
+
+*   We can change the access level (e.g., make a constructor `protected` instead
```

Comment:
Nit: don't think the bullet points actually warrant semi-colon endings. -> periods.

Reply:
Done.